infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Orchestrator.java

```diff
@@ -0,0 +1,78 @@
+package org.corfudb.infrastructure.orchestrator;
+
+import io.netty.channel.ChannelHandlerContext;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.IServerRouter;
+import org.corfudb.infrastructure.orchestrator.workflows.AddNode;
+import org.corfudb.protocols.wireprotocol.CorfuMsgType;
+import org.corfudb.protocols.wireprotocol.CorfuPayloadMsg;
+import org.corfudb.protocols.wireprotocol.orchestrator.AddNodeRequest;
+import org.corfudb.protocols.wireprotocol.orchestrator.OrchestratorRequest;
+import org.corfudb.runtime.CorfuRuntime;
+
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Orchestrator is a service that acts on executing actions that belong to workflow instances.
+ * <p>
+ * Created by Maithem on 10/25/17.
+ */
+
+@Slf4j
+public class Orchestrator {
+
+    final CorfuRuntime runtime;
+
+    /**
+     * The id of this management server.
+     */
+    final UUID id;
+
+    public Orchestrator(CorfuRuntime runtime) {
+        this.runtime = runtime;
+        this.id = UUID.randomUUID();
+    }
+
+    public void handle(CorfuPayloadMsg<OrchestratorRequest> msg,
+                       ChannelHandlerContext ctx, IServerRouter r) {
+
+        OrchestratorRequest req = msg.getPayload();
+        dispatch(req);
+        r.sendResponse(ctx, msg, CorfuMsgType.ORCHESTRATOR_RESPONSE.msg());
+    }
+
+    void dispatch(OrchestratorRequest entry) {
+        CompletableFuture.runAsync(() -> run(entry));
+    }
+
+    private Workflow getWorkflow(OrchestratorRequest req) {
+        switch (req.getWorkFlowType()) {
+            case ADD_NODE:
+                AddNodeRequest addNodeRequest = (AddNodeRequest) req;
+                return new AddNode(addNodeRequest.getEndpoint());
+            default:
+                throw new RuntimeException("Unknown workflow");
+        }
+    }
+
+    void run(OrchestratorRequest request) {
+
+        Workflow workflow = getWorkflow(request);
+
+        log.info("Started workflow {} owner {}", workflow.getName(), id);
+
+
+        for (Action action : workflow.getActions()) {
+            try {
+                log.info("Action {} Status {}", action.getName(), ActionStatus.STARTED);
```

Comment:
This is a strange use of a enum. Shouldn't the ActionStatus enum be assigned somewhere? Ideally override toString in action and just print it.

Reply:
As of now, I'm just emitting logs. Later, the status will be part of the workflow state (i.e. persisted).